### PR TITLE
MG-888 - Update groups SDK tests

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a feature as it updates groups sdk tests.

## What does this do?
This pr:

- Updates tests in `pkg/sdk/go/groups_test.go` to use service mocks.
- Refactors code in `pkg/sdk/go/groups.go` to throw error on missing id in function call before making request.
- Exports the `PageRes` in `pkg/sdk/go/responses.go`

## Which issue(s) does this PR fix/relate to?

- Related Issue #888 

## Have you included tests for your changes?
Yes, the changes are to the tests.

## Did you document any new/modified feature?
No,

### Notes
To be merged after #2195

### CODE REVIEW

**Review by arvindh123 — DISMISSED** [review]

File: pkg/sdk/go/groups_test.go [hunk 1]
```diff
@@ -7,157 +7,260 @@ import (
 	"fmt"
 	"net/http"
 	"net/http/httptest"
+	"strings"
```
Comment (rodneyosodo) [inline pkg/sdk/go/groups_test.go:10]: Add tests for `ListGroupUsers` and `ListGroupChannels`

File: pkg/sdk/go/groups.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/absmach/magistrala/internal/apiutil"
```
Comment (rodneyosodo) [inline pkg/sdk/go/groups.go:12]: Add id check for `getGroups`, `GroupPermissions`, `changeGroupStatus`, `AddUserToGroup`, `RemoveUserFromGroup`, `ListGroupUsers`, `ListGroupUsers`, ``

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Comment by nyagamunene** [comment]
LGTM

File: pkg/sdk/go/groups_test.go (hunk 1 above)
Comment (felixgateru) [reply pkg/sdk/go/groups_test.go:10]: These are covered in users and channels tests respectively. Reorganization of these tests is done by: https://github.com/absmach/magistrala/pull/2195

Comment (felixgateru) [reply pkg/sdk/go/groups_test.go:10]: These are covered in users and channels tests respectively. Reorganization of these tests is done by: https://github.com/absmach/magistrala/pull/2195

File: pkg/sdk/go/groups.go (hunk 2 above)
Comment (felixgateru) [reply pkg/sdk/go/groups.go:12]: The id check for these is done at api level.

**Review by nyagamunene — DISMISSED** [review]
LGTM

File: pkg/sdk/go/groups_test.go (hunk 1 above)
Comment (rodneyosodo) [reply pkg/sdk/go/groups_test.go:10]: @WashingtonKK take note

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**